learn/advanced/pagination.md

```diff
@@ -0,0 +1,117 @@
+
+Paginating requires providing an `offset` that is equal to your `limit` times the page number times: `offset = limit * page number`. We recommend starting page count from 0 instead of 1 when calculating offset values.
+
+For example, if you set `limit` to `20` and want the first page of search results, your `offset` must be 0: `offset = 0 * 20`.
+
+```js
+const results = await index.search("x", { limit: 20, offset: 0 });
+```
+
+If you keep the same `limit` and want the third page of results, `offset` must be `40`: `offset = 2 * 20`.
+
+```js
+const results = await index.search("x", { limit: 20, offset: 40 });
+```
+
+Once a query returns less than your configured `limit`, you have reached the last results page.
+
+It is often helpful to disable navigation buttons when the user cannot move to the "next" or "previous" page. The "previous" button should be disabled whenever your `offset` is 0.
+
+To disable the "next" button, we recommend setting your query's `limit` to the number of displayed results plus one. The extra result indicates that you have at least one item beyond what you can display in a single results page:
+
+```js
+const results = await index.search("x", { limit: 21, offset: 0 })
+
+if (offset === 0 ) {
+  // If offset equals 0, we're on the first results page
+  // and can disable the "previous" buttons
+  document.querySelector("#previous_button").disabled = true;
+} else {
+  document.querySelector("#previous_button").disabled = false;
+}
+
+if (results.hits.length < 21 ) {
+  // If Meilisearch returns 20 items or less, 
+  // we don't have any more results to 
+  // and can disable the "next"
+  document.querySelector("#next_button").disabled = true;
+} else {
+  // If Meilisearch returns exactly 21 results
+  // and our page can only show 20 items at a time,
+  // we have at least one more page with 1 result in it
+  document.querySelector("#next_button").disabled = false;
+}
+```
+
+## Not recommended: Page selection
+
+As its name states, this type of pagination is made of a numbered list of pages accompanied by next and previous buttons.
+
+![placeholder page selection UI](https://vuejsexamples.com/content/images/2018/11/vue-pagination.gif)
+
+This is is a common UI pattern that offers users a significant amount of precision when navigating results.
+
+### Implementation
+
+The general implementation of a page selection interface is similar to our [recommended solution](#recommended-previous-and-next-buttons), but you must also display a page list.
+
+To create the page list, however, you must know the exact number of total results. For example, if you have 100 results and your search result pages contain 10 results each, your selector must show a list of numbers going from 1 to 10.
+
+Since Meilisearch can only give you an estimate of total search results, it can be difficult to implement page selectors when using Meilisearch.
```

Comment:
The other reason is that the estimate can change drastically as you page through results, since Meilisearch will often think you need more results (e.g. for infinite scroll) and look for more.